utils/das_data_sorter.py

```diff
@@ -1,76 +1,170 @@
+        self.ay = self.average_data(data["aY"])
+        self.az = self.average_data(data["aZ"])
+        self.gx = self.average_data(data["gX"])
+        self.gy = self.average_data(data["gY"])
+        self.gz = self.average_data(data["gZ"])
+        self.thermoc = self.average_data(data["thermoC"])
+        self.thermof = self.average_data(data["thermoF"])
+        self.pot = self.average_data(data["pot"])
+        self.cadence = self.average_data(data["cadence"])
+        self.power = self.average_data(data["power"])
+        self.reed_velocity = self.average_data(data["reed_velocity"])
+        self.reed_distance = self.average_data(data["reed_distance"])
+
+        self.print_output_file()
+
+    def convert_time(self, milliseconds:D) -> D:
+        '''Returns the conversion of the time data points from milliseconds to the specified time unit, depending on the 
+        argument passed by args.unit.
+        
+        args.unit accepts only seconds/s and minutes/m as arguments. 
+        '''
+        if args.unit == "seconds" or args.unit == "s":
+            milliseconds = milliseconds/1000
+        elif args.unit == "minutes" or args.unit == "m":
+            milliseconds = milliseconds/1000/60
+        else:
+            raise ValueError("Argument only accepts either seconds/s or minutes/m")
+        
+        # sets the groups of indexes to use for averaging in future
+        self.indexes = self.group_index(milliseconds)
+
+        return range(1,len(self.indexes)+1)
+
+    def group_index(self, time:D) -> D:
+        '''Returns a universal array of arrays of indexes, based on the specified time unit. Each array represents the indexes 
+        of the data points within the same time interval. (eg. 1123ms and 1748ms are in the same time interval for seconds, 
+        but 2453ms isn't)
+
+        When given a valid value for args.unit, function will group the indexes within the same time interval in an array.
+        Once all the indexes of a time interval has been accounted for, the array of indexes will be pushed into a
+        universal array so that each element represents the data points of one singular time interval.
+        '''
+        universal_array = []
+        index_array = []
+        previous_time = 0
+
+        for i in range(len(time)):
+            if time[i] > previous_time:
+                # if true, then time at time[i] is the next second/minute
+                if previous_time != 0:
+                    # appends index array into universal array, but ignores the first iteration
+                    universal_array.append(index_array)
+
+                previous_time = ceil(time[i])
+                index_array = [] # recreates new array if previous time has changed
+            index_array.append(i) 
+
+        # pushes the final array when for loop has completed
+        universal_array.append(index_array)
+
+        return universal_array
+
+    def mean(self, data_array:list) -> int:
```

Comment:
Input type is pd.Series, return type will be `float`.

Although, `list`s will still work in this function. If you wanted to be really fancy you could do `Union(pd.Series, list)` from the typing module, but that isn't necessary as we don't pass this function `list`s from anywhere.